src/Contexts/section.actions.jsx

```diff
@@ -0,0 +1,417 @@
+				payload: sectionId
+			})
+		} else {
+			dispatch({
+				type: SAVING_DOCVALUE_FAILED,
+				payload: sectionId
+			})
+		}
+	})
+
+	resolved.forEach((res, i) => {
+		// keep care of adding or removing something, ...section is to generate hash and it should include
+		// everything witch is part of section, compared to database
+		const {changed, ...section} = sections[i]
+		const sectionId = section._id
+		if(res.status === 'fulfilled'){
+			dispatch({
+				type: DOCVALUE_SAVED,
+				payload: sectionId
+			})
+			saveSectionData({
+				...section,
+				timestamp: res.value.updatedAt || res.value.instertedAt,
+				docHash: hashes[i],
+				savedToBackend: true
+			})
+		} else {
+			dispatch({
+				type: SAVING_DOCVALUE_FAILED,
+				payload: sectionId
+			})
+			saveSectionData({
+				section,
+				docHash: hashes[i],
+				savedToBackend: false
+			})
+		}
+	})
+}
+
+/**
+ * Saves all data to server, that are markes in the changed Set
+ */
+export const saveSections = () => async ({dispatch, state}) => {
+
+	dispatch({
+		type: SAVING_SECTIONS
+	})
+
+	const { sections, lesson } = state
+
+	const hashes = []
+	// keep care of adding or removing something, ...section is to generate hash and it should include
+	// everything witch is part of section, compared to database
+	const proms = sections.map(({changed, hash, timestamp, ...section}, index) => {
+		const changes = {}
+
+		if(changed.has('docValue')){
+			// docValue is removed from changed Set to not filter it afterwards
+			// saveSectionDocValue, should not check it, but set it again if it failed
+			changed.delete('docValue')
+			dispatch(saveSectionDocValue({ignoreChangedSet: index}))
+		}
+
+		if(changed.size === 0) return Promise.resolve('nothing to save')
+
+		changed.forEach(key => {
+			if(Object.prototype.hasOwnProperty.call(section, key)){
+				changes[key] = section[key]
+			}
+		})
+
+		const prom = editorWS.emit(
+			'patch',
+			`lesson/${lesson._id}/sections`,
+			section._id,
+			changes
+		)
+
+		// meaby a hash should be part of the server request, but not implemented write now
+		hashes[index] = generateHash(section)
+
+		return prom
+	})
+
+	const resolved = await Promise.allSettled(proms)
+
+	resolved.forEach((res, i) => {
+		// keep care of adding or removing something, ...section is to generate hash and it should include
+		// everything witch is part of section, compared to database
+		const {changed, ...section} = sections[i]
+		const sectionId = section._id
+
+		if(res.status === 'fulfilled'){
+			if(res.value !== 'nothing to save'){
+				dispatch({
+					type: SECTION_SAVED,
+					payload: sectionId
+				})
+			}
+			saveSectionData({
+				...section,
+				// timestamp: res.value.updatedAt || res.value.instertedAt,
+				hash: hashes[i],
+				savedToBackend: true
+			})
+		} else {
+			dispatch({
+				type: SAVING_SECTION_FAILED,
+				payload: sectionId
+			})
+			saveSectionData({
+				section,
+				hash: hashes[i],
+				savedToBackend: false
+			})
+		}
+	})
+
```

Comment:
Teilweise wird hier sehr viel Platz beansprucht der es nicht übersichtlicher macht. 
```
line break
line break <-- weg 1
/** <-- weg 2
 * Sets docValue
 *  <-- weg 3
 * @param 
```
Kann man gut um 3 Zeilen einkürzen. Das sind auf die ganzen Änderung in allen Dokumenten sehr viele lines.